direct/core/src/main/java/cz/o2/proxima/direct/commitlog/LogObserver.java

```diff
@@ -27,6 +28,53 @@
 @Stable
 public interface LogObserver extends Serializable {
 
+  /**
+   * Create a {@link LogObserver} that handles retries on {@link Exception Exceptions} thrown during
+   * processing. If retries are exhausted, the last caught Exception is rethrown.
+   *
+   * @param name name of the consumer
+   * @param numRetries number of retries that should be attempted
+   * @param delegate the {@link LogObserver} to handle to processing
+   * @return {@link LogObserver} that will retry processing
+   */
+  static LogObserver withNumRetriedExceptions(String name, int numRetries, LogObserver delegate) {
```

Comment:
`withMaxExceptions` / `withMaxRetriedExceptions`?

Reply:
I know that name allows easier debugging, but can we somehow work around the need of passing it here? Eg. by introduciong some kind of `NameAwareLongObserver` interface, that commit log / .. / .. can use to set the name automatically?

Reply:
That would actually not help. We would have to rework the entire process of how we call `observe` on a `CommitLogReader`. The "workaround" this, was how `RetryableLogObserver.start()`, which we removed. We therefore have no way of not-specifing the name twice.